ng-dev/utils/git/github-urls.ts

```diff
@@ -20,7 +20,8 @@ export const GITHUB_TOKEN_GENERATE_URL = 'https://github.com/settings/tokens/new
 /** Adds the provided token to the given Github HTTPs remote url. */
 export function addTokenToGitHttpsUrl(githubHttpsUrl: string, token: string) {
   const url = new URL(githubHttpsUrl);
-  url.username = token;
+  url.password = token;
+  url.username = '_';
```

Comment:
Out of curiosity: how did our merge tooling etc. work without this before? I assume (but you know better with your testing in `main-test`) because in those cases we deal with a repo that is actively used and where a user is already set in the repo? and for actions we now this need because we have set `persist-credentials: false`, so there is no actual user stored in credentials (like it would be local for the caretaker running yarn ng-dev pr merge)?

I think we should try to capture this here in the PR, or best in a comment..

Reply:
Agreed we should document this somewhere.  I am not sure where the best place to include it is though because I think a comment is the best option, but I don't know where that comment would even go.

The answer to why it worked before is that everyone running the commands were always running things from an environment that already had authentication set up, so if the authentication didn't work via the `remote` url, it just used the git cached credentials.

Reply:
I think this comment is also sufficient for future reasoning

Reply:
Added that block into the commit message as well.